vllm/v1/core/specialized_manager.py

```diff
@@ -164,17 +334,27 @@ def remove_skipped_blocks(self, blocks: list[KVCacheBlock],
                 break
             removed_blocks.append(blocks[i])
             blocks[i] = self._null_block
-        return removed_blocks
+        self.block_pool.free_blocks(removed_blocks)
+
+    @override
```

Comment:
QQ: Is this necessary?

Reply:
“override” is not necessary but I want to have it to help the type checker.

Reply:
Also I think `get_num_common_prefix_blocks` in `SingleTypeKVCacheManager` should move to `FullAttentionManager`. The logic is specific to full attention rather than general.

Reply:
Agree with moving `get_num_common_prefix_blocks` to `FullAttentionManager`.